token/cli/src/config.rs

```diff
@@ -134,34 +134,45 @@ impl<'a> Config<'a> {
 
         let default_keypair = cli_config.keypair_path.clone();
 
-        let default_signer: Arc<dyn Signer> = {
+        let default_signer: Option<Arc<dyn Signer>> = {
             if let Some(owner_path) = matches.value_of("owner") {
                 signer_from_path_with_config(matches, owner_path, "owner", wallet_manager, &config)
+                    .ok()
             } else {
-                signer_from_path_with_config(
+                match signer_from_path_with_config(
                     matches,
                     &default_keypair,
                     "default",
                     wallet_manager,
                     &config,
-                )
+                ) {
+                    Ok(signer) => Some(signer),
+                    Err(e) => {
+                        if std::fs::metadata(&default_keypair).is_ok() {
+                            eprintln!("error: {}", e);
+                            exit(1);
+                        } else {
+                            None
+                        }
+                    }
+                }
             }
         }
-        .map(Arc::from)
-        .unwrap_or_else(|e| {
-            eprintln!("error: {}", e);
-            exit(1);
-        });
+        .map(Arc::from);
 
-        let fee_payer = matches
+        let fee_payer: Option<Arc<dyn Signer>> = matches
             .value_of("fee_payer")
-            .map_or(Ok(default_signer.clone()), |path| {
-                signer_from_path(matches, path, "fee_payer", wallet_manager).map(Arc::from)
+            .map(|path| {
+                Arc::from(
+                    signer_from_path(matches, path, "fee_payer", wallet_manager).unwrap_or_else(
+                        |e| {
+                            eprintln!("error: {}", e);
+                            exit(1);
+                        },
+                    ),
+                )
             })
-            .unwrap_or_else(|e| {
-                eprintln!("error: {}", e);
-                exit(1);
-            });
+            .or(default_signer.clone());
```

Comment:
I think clippy wants this to be
```suggestion
            .or_else(|| default_signer.clone());
```
You can check this locally by running `cargo +nightly-2022-04-01 clippy -Zunstable-options --all-targets  -- --deny=warnings --deny=clippy::integer_arithmetic`